specifications/client/verifying_client.md

```diff
@@ -0,0 +1,125 @@
+# Diem Verifying Client Spec
+
+THIS IS A WORK IN PROGRESS
+
+Links:
+
+* https://github.com/novifinancial/libra-specs/blob/master/client/client.md#updatetolatestledger-requests
+* https://github.com/novifinancial/libragogo/blob/master/client_specification.md#submittransaction-request
+* https://github.com/diem/diem/blob/master/json-rpc/json-rpc-spec.md
+
+## Data serialisation
+
+Serialisation is done using BCS (https://docs.rs/bcs/0.1.2/bcs/) it should be implemented in order to communicate with the verifying API endpoint, but also in order to verify the proofs, since these assume BCS encoded data sometimes.
+
+## Verifying signatures
+
+We refer to the [Diem Crypto spec](https://github.com/diem/diem/blob/master/specifications/crypto/README.md#signing-and-verifying) for more details regarding how the verification is to be handled. But this requires implementing BCS encoding.
+
+## JSON-RPC Endpoint
+
+The clients must query a full node, this is done using the Full Node’s JSON-RPC interface.
+Currently verification is only supported by the following (experimental) APIs:
+
+* get_state_proof
+* get_account_state_with_proof
+* get_transactions_with_proofs
+* get_events_with_proofs
+
+Notice that these endpoints should be enough to run a light client that would have fully verified proofs, without any need to trust any given full node.
+
+You can find the JSON-RPC specification here:  https://github.com/diem/diem/blob/master/json-rpc/json-rpc-spec.md
+
+## Alternative to verifying proofs
+
+Instead of verifying proofs, a client might want to :
+
+* query randomly multiple nodes and only accept an answer if it is the one given by the majority of the nodes
```

Comment:
This is interesting as it will have different security implications. It would be good to note that, and explain what sorts of security properties you're getting from the approach in this spec VS the approach mentioned here